lib/archethic/governance/code.ex

```diff
@@ -153,28 +153,43 @@ defmodule Archethic.Governance.Code do
   """
   @spec applicable_proposal?(Proposal.t()) :: boolean()
   def applicable_proposal?(
-        %Proposal{changes: changes, address: address},
+        proposal,
         src_dir \\ @src_dir
       ) do
+    res = apply_diff(proposal, src_dir, false)
+    match?({_, 0}, res)
+  end
+
+  defp apply_diff(
+         %Proposal{changes: changes, address: address},
+         src_dir,
+         persist?
+       ) do
     random = :crypto.strong_rand_bytes(4) |> Base.encode16()
     prop_file = Path.join(System.tmp_dir!(), "prop_#{random}_#{Base.encode16(address)}")
```

Comment:
Why use some randomness along the address, as we can only have one proposal per address ?

Reply:
I didn't change this part, but I think it was helpful in local dev but you are right we don't need to have the randomness.

Reply:
Ok